features-sdk/step_definitions/security-steps.js

```diff
@@ -1,23 +1,55 @@
 const
-  { Given } = require('cucumber'),
-  should = require('should');
+  should = require('should'),
+  {
+    Given,
+    Then
+  } = require('cucumber');
 
-Given('I create a profile {string} with the following policies:', function (profileId, dataTable) {
-  let policies = this.parseObject(dataTable);
+Given('I create a profile {string} with the following policies:', async function (profileId, dataTable) {
+  const data = this.parseObject(dataTable);
+  const policies = [];
+  for (const [roleId, restrictedTo] of Object.entries(data)) {
+    policies.push({ roleId, restrictedTo });
+  }
+  this.props.result = await this.sdk.security.createProfile(profileId, {policies});
+});
+
+Given('I create a role {string} with the following API rights:', async function (roleId, dataTable) {
+  const controllers = this.parseObject(dataTable);
+  this.props.result = await this.sdk.security.createRole(roleId, { controllers }, { refresh: 'wait_for' });
+});
 
-  policies = Array.isArray(policies) ? policies : [policies];
+Then(/I can( not)? delete the role "(.*?)"/, async function (not, roleId) {
 
-  return this.sdk.security.createProfile(profileId, { policies });
+  try {
+    await this.sdk.security.deleteRole(roleId, { refresh: 'wait_for' });
+  } catch (e) {
```

Comment:
Nitpicking: the `catch` should be on the next line conforming to our coding style standard
```
try {

}
catch (error) {

}
```